sdk/storage/Azure.Storage.Files.DataLake/api/Azure.Storage.Files.DataLake.netstandard2.0.cs

```diff
@@ -258,6 +264,33 @@ public partial class DataLakeUriBuilder
 }
 namespace Azure.Storage.Files.DataLake.Models
 {
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public partial struct AccessControlRecursiveChangeFailure
+    {
+        private object _dummy;
+        private int _dummyPrimitive;
+        public string ErrorMessage { get { throw null; } }
+        public bool IsDirectory { get { throw null; } }
+        public string Name { get { throw null; } }
+    }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public partial struct AccessControlRecursiveChangeOptions
+    {
+        public int? BatchSize { get { throw null; } set { } }
+    }
+    public partial class AccessControlRecursiveChanges : Azure.Storage.Files.DataLake.Models.AccessControlRecursiveChangesSummary
+    {
+        internal AccessControlRecursiveChanges() { }
+        public System.Collections.Generic.IEnumerable<Azure.Storage.Files.DataLake.Models.AccessControlRecursiveChangeFailure> FailedEntries { get { throw null; } }
+    }
+    public partial class AccessControlRecursiveChangesSummary
```

Comment:
I don't love the suffix `Summary` because I don't think we've used that anywhere else, but with the class hierarchy it makes sense.  Are there any customer scenarios for polymorphing over this type and should we actually derive `AccessControlRecursiveChanges` from it?  (You don't have to solve this before checking in.)

Reply:
`AccessControlRecursiveChanges` could contain (composition) `AccessControlRecursiveChangesSummary with partial counters (I think it's better idea than inheritance after giving it more thought). 
Regarding `Summary`. Previously we had `Result` and `PartialResult`. `Summary` seems to be valid term for both partial and final result. How else would we name bag containing aggregated counters?

Reply:
Actually - if you use composition you could provide both the current counters and the aggregate counters thus far which might be interesting to customers.  What if we had

```C#
public class AccessControlChangeCounters { long ChangedDirectoriesCount; /* ... */ }
public class AccessControlChanges
{
    public IEnumerable<AccessControlChangeFailure> RequestFailures { get; }
    public AccessControlChangeCounters RequestCounters { get; }
    public AccessControlChangeCounters AggregateCounters { get; }
}
```

All those names aren't quite right, but hopefully it conveys the idea.

Reply:
Great. I'll make it so. Thank you.